docs/zero_to_hero_guide/quickstart.md

```diff
@@ -0,0 +1,185 @@
+# Llama Stack Quickstart Guide
+
+This guide will walk you through setting up an end-to-end workflow with Llama Stack, enabling you to perform text generation using the `Llama3.2-3B-Instruct` model. Follow these steps to get started quickly.
+
+If you're looking for more specific topics like tool calling or agent setup, we have a [Zero to Hero Guide](#next-steps) that covers everything from Tool Calling to Agents in detail. Feel free to skip to the end to explore the advanced topics you're interested in.
+
+## Table of Contents
+1. [Setting up](#Setting-up)
+2. [Build, Configure, and Run Llama Stack](#build-configure-and-run-llama-stack)
+3. [Testing with `curl`](#testing-with-curl)
+4. [Testing with Python](#testing-with-python)
+5. [Next Steps](#next-steps)
+
+---
+
+
+
+## Setting up
+
+### 1. Prerequisite
+
+Ensure you have the following installed on your system:
+
+- **Conda**: A package, dependency, and environment management tool.
+
+
+### 2. Installation
+The `llama` CLI tool helps you manage the Llama Stack toolchain and agent systems.
+
+
+```bash
+pip install llama-stack
+```
+
+After installation, the `llama` command should be available in your PATH.
+
+### 3. Download Llama Models
+
+Download the necessary Llama model checkpoints using the `llama` CLI:
+
+```bash
+llama download --model-id Llama3.2-3B-Instruct
+```
+
+Follow the CLI prompts to complete the download. You may need to accept a license agreement. Obtain an instant license [here](https://www.llama.com/llama-downloads/).
+
+---
+
+## Build, Configure, and Run Llama Stack
+
+### 1. Build the Llama Stack Distribution
+
+We will default to building the `meta-reference-gpu` distribution due to its optimized configuration tailored for inference tasks that utilize local GPU capabilities effectively. If you have limited GPU resources, prefer using a cloud-based instance or plan to run on a CPU, you can explore other distribution options [here](https://llama-stack.readthedocs.io/en/latest/getting_started/index.html#decide-your-inference-provider).
+
+```bash
+llama stack build --template meta-reference-gpu --image-type conda
+```
+
+
+### 2. Run the Llama Stack Distribution
+> Launching a distribution initializes and configures the necessary APIs and Providers, enabling seamless interaction with the underlying model.
+
+Start the server with the configured stack:
+
+```bash
+cd llama-stack/distributions/meta-reference-gpu
+llama stack run ./run.yaml
```

Comment:
Also seems llama guard has been activated which seems expecting user to download before hand.

```
llama stack run ./run.yaml
Using config `run.yaml`
Resolved 13 providers
 inner-inference => meta-reference-inference
 inner-inference => meta-reference-safety
 models => __routing_table__
 inference => __autorouted__
 inner-safety => meta0
 inner-memory => meta0
 shields => __routing_table__
 safety => __autorouted__
 memory_banks => __routing_table__
 memory => __autorouted__
 agents => meta0
 telemetry => meta0
 inspect => __builtin__

Loading model `Llama3.2-3B-Instruct`
...
AssertionError: Could not find checkpoints in: /home/hamidnazeri/.llama/checkpoints/Llama-Guard-3-1B. Please download model using `llama download --model-id Llama-Guard-3-1B`
```

Reply:
thanks for pointing it out, fixed